### ISSUE TITLE
Allow passing a block to Expectation#returns

### ISSUE THREAD

**Issue by Hubro** [freerange/mocha#230]
Instead of this:

```
class SomethingTest
  def setup
    @subject = Something.new
    @test_data = fetch_test_data
  end

  def test_foo
    @subject.stub(:fetch_over_network).returns(test_data["foo"])
    # ...
  end

  def test_bar
    @subject.stub(:fetch_over_network).returns(test_data["bar"])
    # ...
  end

  def test_baz
    @subject.stub(:fetch_over_network).returns(test_data["baz"])
    # ...
  end
end
```

I would like to be able to do this:

```
class SomethingTest
  def setup
    @subject = Something.new
    @subject.stub(:fetch_over_network).returns { |input| test_data[input] }
  end

  # ...
end
```

I yearn to be able to just pass a block to `returns` and decide the return value based on the input arguments. It would save so much boilerplate in situations like this...

Additionally, it happens too often that I forget to stub a method, or I write a typo (`:fetch_over_netrwok`) and end up actually fetching something over the network every time my test suite runs.

To summarize, I feel like implementing this would be very DRY of you, and would in some situations drastically reduce the amount of boilerplate code in tests.

**Comment by booleanbetrayal** [comment]
:+1:

**Comment by floehopper** [comment]
Closing in favour of #224.
